File: Exercise/Medium/10. Column-wise Matrix Fill/Code/Column-wise Matrix Fill.c
#include <stdio.h>

void fullMatrix(int n, int arr[n][n]) {
    int num = 1;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            arr[i][j] = num++;
        }
    }
}

void printMatrix(int n, const int arr[n][n]) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("%d ", arr[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int n;
    scanf("%d", &n);

    int arr[n][n];

    fullMatrix(n, arr);
    printMatrix(n, arr);

    return 0;
}

File: Exercise/Medium/10. Column-wise Matrix Fill/Code/Column-wise Matrix Fill.java
import java.util.Scanner;

public class Main {
    static void fullMatrix(int n, int[][] arr) {
        int num = 1;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                arr[i][j] = num++;
            }
        }
    }

    static void printMatrix(int n, int[][] arr) {
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                System.out.print(arr[i][j] + " ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        int[][] arr = new int[n][n];

        fullMatrix(n, arr);
        printMatrix(n, arr);
    }
}



